src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AssigningSymbolAndItsMemberInSameStatement.Fixer.cs

```diff
@@ -0,0 +1,139 @@
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(AssigningSymbolAndItsMemberInSameStatement.RuleId);
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            // For something like `a.x = a = b;`, we offer 3 code fixes:
+            // First:
+            //     a = b;
+            //     a.x = b;
+            // Second:
+            //     a = b;
+            //     a.x = a;
+            // Third: (Not currently implemented)
+            //     var temp = a;
+            //     a = b;
+            //     temp.x = b;
+
+            var title = MicrosoftCodeQualityAnalyzersResources.AssigningSymbolAndItsMemberInSameStatementTitle;
+            context.RegisterCodeFix(new MyCodeAction(title,
+                 async ct => await SplitAssignmentFirstOption(context.Document, context.Span, ct).ConfigureAwait(false),
+                 equivalenceKey: title + "0"),
+            context.Diagnostics);
+            context.RegisterCodeFix(new MyCodeAction(title,
+                 async ct => await SplitAssignmentSecondOption(context.Document, context.Span, ct).ConfigureAwait(false),
+                 equivalenceKey: title + "1"),
+            context.Diagnostics);
+            return Task.CompletedTask;
+        }
+
+        private static async Task<Document> SplitAssignmentFirstOption(Document document, TextSpan span, CancellationToken cancellationToken)
+        {
+            // This method splits `a.x = a = b` to:
+            // a = b;
+            // a.x = b;
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            // a.x = a = b;
+            var parentAssignment = root.FindNode(span).Parent;
+
+            if (!TryGetAssignmentExpressionParts(parentAssignment, out _, out var right) ||
+                !TryGetAssignmentExpressionParts(right, out _, out var rightOfRight))
+            {
+                return document;
+            }
+
+            // a = b;
+            right = GetExpressionFromAssignment(right).WithTriviaFrom(parentAssignment.Parent);
+
+            // a.x = b;
+            var firstEqualsLastAssignment = GetExpressionFromAssignment(GetAssignmentWithRight(parentAssignment, rightOfRight));
+
+            root = root.ReplaceNode(parentAssignment.Parent, new[] { right, firstEqualsLastAssignment });
+            return document.WithSyntaxRoot(root);
+        }
+
+        private static async Task<Document> SplitAssignmentSecondOption(Document document, TextSpan span, CancellationToken cancellationToken)
+        {
+            // This method splits `a.x = a = b` to:
+            // a = b;
+            // a.x = a;
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            // a.x = a = b;
+            var parentAssignment = root.FindNode(span).Parent;
```

Comment:
Why don't you do the cast here? Instead of having a direct-cast directly in the various sub-method called.

Reply:
@Evangelink Yes I should do that.
I was writing this first as an abstract base class for both C# and VB before I realize that it's not applicable to VB.

Reply:
I think I'll probably need to change the way this works as it currently fails when the assignment have 4 members.